client.go

```diff
@@ -959,8 +959,17 @@ func clientDoDeadline(req *Request, resp *Response, deadline time.Time, c client
 	// Without this 'hack' the load on slow host could exceed MaxConns*
 	// concurrent requests, since timed out requests on client side
 	// usually continue execution on the host.
+
+	var cleanup int32
+	atomic.StoreInt32(&cleanup, 0)
```

Comment:
Since the other gorountine hasn't started here and the default value of an `int32` is `0` you don't need this line at all.

Reply:
@amezghal if you just remove this line then it's perfect and I can merge it.